File: python/contains_duplicate.py
# LeetCode 217 - Contains Duplicate
# Difficulty: Easy
# Time Complexity: O(n)
# Space Complexity: O(n)

from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen_numbers = set()
        for current_number in nums:
            if current_number in seen_numbers:
                return True
            seen_numbers.add(current_number)
        return False

File: kbhatiya999/java/ContainsDuplicate.java
// LeetCode 217 - Contains Duplicate
// Difficulty: Easy
// Time Complexity: O(n)
// Space Complexity: O(n)

import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }
}
